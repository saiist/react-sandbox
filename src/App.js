function App({ newwer }) {
  if (newwer) {
    return (
      <div className="App">
        <header className="App-header">新規登録</header>
      </div>
    );
  }
  return (
    <div className="App">
      <header className="App-header2">更新</header>
    </div>
  );
}

export default App;
