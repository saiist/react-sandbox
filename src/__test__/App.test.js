import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App.js スナップショット', () => {
  it('新規登録の場合', () => {
    const tree = renderer.create(<App newwer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('更新の場合', () => {
    const tree = renderer.create(<App newwer={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
