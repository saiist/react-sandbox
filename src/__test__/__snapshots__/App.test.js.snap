// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js スナップショット 新規登録の場合 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    新規登録
  </header>
</div>
`;

exports[`App.js スナップショット 更新の場合 1`] = `
<div
  className="App"
>
  <header
    className="App-header2"
  >
    更新
  </header>
</div>
`;
